app/controllers/transition_notes_controller.rb

```diff
@@ -0,0 +1,47 @@
+class TransitionNotesController < ApplicationController
+  before_action :authorize!
+
+  def authorize!
+    student = Student.find(params[:student_id])
+    educator = current_educator
+    raise Exceptions::EducatorNotAuthorized unless educator && educator.is_authorized_for_student(student)
+    raise Exceptions::EducatorNotAuthorized unless educator && educator.labels.include?('k8_counselor')
+  end
+
+  def create
+    transition_note = TransitionNote.new(transition_note_params.merge({
+      educator_id: current_educator.id,
+      recorded_at: Time.now
+    }))
+
+    if transition_note.save
+      render json: { result: 'ok', is_restricted: transition_note.is_restricted }
+    else
+      render json: { errors: transition_note.errors.full_messages }, status: 422
+    end
+  end
+
+  def update
+    transition_note = TransitionNote.find_by_id(params[:id])
+
+    transition_note.text = transition_note_params[:text]
+
+    if transition_note.save
+      render json: { result: 'ok', is_restricted: transition_note.is_restricted }
+    else
+      render json: { errors: transition_note.errors.full_messages }, status: 422
+    end
+  end
+
+  private
+
+    def transition_note_params
+      params.require(:transition_note).permit(
```

Comment:
Is this a security hole?  I can't tell if it's guaranteed that the `student_id` param here is the same as the `Student.find(params[:student_id]` in the before filter.  It'll be safe as long as both places read the same value, or if you check they match and raise if not.

Reply:
Actually I don't think `transition_note_params` can be the same for create / update.  We can't allow an `id` field in `create`.